libglusterfs/src/client_t.c

```diff
@@ -377,59 +377,31 @@ gf_client_unref(client_t *client)
     }
 }
 
+/* value (assumed to be passed NULL) is returned if the key was found.
+   ret value >= 0 - the location it is found (value is non NULL)
+   or the 1st free slot (in case value is NULL)
+   or -1 in case no free slots found
+*/
 static int
-__client_ctx_get_int(client_t *client, void *key, void **value)
-{
-    int index = 0;
-    int ret = 0;
-
-    for (index = 0; index < GF_CLIENTCTX_INITIAL_SIZE; index++) {
-        if (client->scratch_ctx[index].ctx_key == key)
-            break;
-    }
-
-    if (index == GF_CLIENTCTX_INITIAL_SIZE) {
-        ret = -1;
-        goto out;
-    }
-
-    if (value)
-        *value = client->scratch_ctx[index].ctx_value;
-
-out:
-    return ret;
-}
-
-static int
-__client_ctx_set_int(client_t *client, void *key, void *value)
+__client_ctx_get_int(client_t *client, void *key, void *value)
 {
     int index = 0;
-    int ret = 0;
     int set_idx = -1;
 
     for (index = 0; index < GF_CLIENTCTX_INITIAL_SIZE; index++) {
         if (!client->scratch_ctx[index].ctx_key) {
             if (set_idx == -1)
                 set_idx = index;
-            /* don't break, to check if key already exists
+            /* don't break, check if key already exists
                further on */
-        }
-        if (client->scratch_ctx[index].ctx_key == key) {
-            set_idx = index;
-            break;
+        } else if (client->scratch_ctx[index].ctx_key == key) {
+            value = client->scratch_ctx[index].ctx_value;
```

Comment:
```suggestion
            *value = client->scratch_ctx[index].ctx_value;
```

Reply:
note to self - don't send patches on midnight.

Fixed.